src/main/kotlin/org/rust/cargo/project/workspace/CargoWorkspace.kt

```diff
@@ -78,10 +77,10 @@ class CargoWorkspace private constructor(
 
     private val targetByCrateRootUrl = packages.flatMap { it.targets }.associateBy { it.crateRootUrl }
 
-    fun findCrateByName(normName: String): Target? =
-        packages
-            .mapNotNull { it.libTarget }
-            .find { it.normName == normName }
+    fun findCrateByName(normName: String, inPackage: Package? = null): Target? =
```

Comment:
Hm, that's indeed the case that Cargo itself tracks deps on the package granularity, so it's totally OK to do the same.

There is a bug filed to support "binary only" dependencies in Cargo though, but it probably won't be fixed in the nearish future.